### DISCUSSION TITLE
How to load an image2 format from FFMPEG?

### DISCUSSION THREAD

**Discussion by intermension** [SixLabors/ImageSharp#1512] [Q&A] [answered]
I'm creating thumbnails from video using FFMPEG pipes. When piped the image is returned an image2 format which i believe to be a YUV file of some kind. In .NET 47 I can do the following to load the image for subsequent processing:

`using (var img = Image.FromStream(thumbStream))`

and the conversion is automatically handled but in ImageSharp it fails:
```
using (Image image = Image.Load(thumbStream))
                    
Image cannot be loaded. Available decoders:
 - GIF : GifDecoder
 - TGA : TgaDecoder
 - PNG : PngDecoder
 - BMP : BmpDecoder
 - JPEG : JpegDecoder
```
 Does anyone know if there is something simple i can do to resolve this?

**Comment by antonfirsov** [comment, marked as answer]
ImageSharp only has builtin support for the formats listed in the error message.

Maybe you can use [FFMediaToolkit](https://github.com/radek-k/FFMediaToolkit) to load the stream and get the frame as `ImageData` as described here:
https://github.com/radek-k/FFMediaToolkit#code-samples

Then convert it to an ImageSharp image like this:
https://github.com/radek-k/FFMediaToolkit#usage-details

**Reply by JimBobSquarePants** [reply]
That's cool they have a dedicated example for us!

**Reply by intermension** [reply]
That's helpful. Thanks.

**Comment by JimBobSquarePants** [comment]
I’m curious... were you originally using System.Drawing? If so, I’m wondering if it could simply be a stream position issue?

**Reply by intermension** [reply]
Yes I was. I thought the same but setting position back to 0 did not resolve the issue.

**Reply by JimBobSquarePants** [reply]
Thanks... I wonder what the image format of the stream contents is?

**Reply by intermension** [reply]
It is actually a JFIF:

```
Header (from winhex)
FFD8FFE000104A46494600010200000100010000FFFE00104C61766335382E33352E31303000FFDB0043
```

So I'm not sure what ImageSharp uses to determine compatibility and if i save the stream to disk then image editors can use it.

**Reply by JimBobSquarePants** [reply]
Ok.... That's weird... We should be able to detect it.

https://github.com/SixLabors/ImageSharp/blob/eab04e41a5f10b70131d0c52caaf2f10f5969d0f/src/ImageSharp/Formats/Jpeg/JpegImageFormatDetector.cs#L58-L60

**Reply by antonfirsov** [reply]
Out of curiosity: how is JFIF/JPEG related to FFMPEG?
